src/containers/AuthenticationPage/AuthenticationPage.js

```diff
@@ -136,10 +172,88 @@ export class AuthenticationPageComponent extends Component {
       submitSignup(params);
     };
 
-    const formContent = (
+    const handleSubmitConfirm = values => {
+      const { idpToken, email, firstName, lastName, idpId } = this.state.authInfo;
+      const { email: newEmail, firstName: newFirstName, lastName: newLastName } = values;
+
+      // Pass email, fistName or lastName to Flex API only if user has edited them
+      // sand they can't be fetched directly from idp provider (e.g. Facebook)
+
+      const authParams = {
+        ...(newEmail !== email && { email: newEmail }),
+        ...(newFirstName !== firstName && { firstName: newFirstName }),
+        ...(newLastName !== lastName && { lastName: newLastName }),
+      };
+
+      submitSinguoWithIdp({
+        idpToken,
+        idpId,
+        ...authParams,
+      });
+    };
+
+    const authWithFacebook = () => {
+      const baseUrl = apiBaseUrl();
+
+      if (from) {
+        window.location.href = `${baseUrl}/api/auth/facebook?from=${from}`;
+      } else {
+        window.location.href = `${baseUrl}/api/auth/facebook`;
+      }
+    };
+
+    const idp = this.state.authInfo ? this.state.authInfo.idpId : null;
+
+    // showSocialLoginsForTests makes sure that snapshot shows a correct condition
+    // if that prop is not given, show social logins if Facebook id set in env variables
+    const showSocialLogins = showSocialLoginsForTests
+      ? showSocialLoginsForTests
+      : !!config.facebookAppId;
+
+    const socialLoginButtonsMaybe = showSocialLogins ? (
+      <div className={css.idpButtons}>
+        <SocialLoginButton onClick={() => authWithFacebook()}>
+          <span className={css.buttonIcon}>
+            <svg width="20" height="20" xmlns="http://www.w3.org/2000/svg">
+              <path
+                d="M7.89.214C4.055 1.047 1.005 4.13.205 7.947c-.734 3.45.533 7.283 3.166 9.6.967.85 3.2 2.033 4.15 2.183l.617.1v-6.883H5.806v-3h2.283l.083-1.633c.134-2.417.717-3.534 2.3-4.25.617-.284 1.034-.35 2.3-.334.85.017 1.617.084 1.7.134.1.05.167.7.167 1.433v1.317h-.983c-1.484 0-1.75.283-1.817 1.983l-.067 1.35h1.45c1.284 0 1.434.033 1.35.283-.05.167-.133.667-.2 1.134-.216 1.55-.25 1.583-1.483 1.583h-1.083V19.914l.866-.234c1.684-.433 2.984-1.216 4.4-2.633 2.067-2.067 2.9-4.1 2.9-7.017 0-3.166-1.2-5.75-3.616-7.766C14.106.38 10.772-.42 7.889.214z"
+                fill="#1877F2"
+                fillRule="nonzero"
+              />
+            </svg>
```

Comment:
It's probably better to make SVGs always as separate subcomponents (it could be inside this directory). 
The reasoning has been that people are not familiar with SVG syntax - having them in a separate component makes bug hunting a bit more straightforward (and people are not so tempted to tamper this unfamiliar code/colors/attributes).

Reply:
It's also easier to add comments like: 
"// This is Facebook's logo, you are not allowed to change its color"